hadoop-hdds/managed-rocksdb/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedRocksDB.java

```diff
@@ -78,4 +86,20 @@ public static ManagedRocksDB open(
         RocksDB.open(path, columnFamilyDescriptors, columnFamilyHandles)
     );
   }
+
+  /**
+   * Delete liveMetaDataFile from rocks db using RocksDB#deleteFile Api.
+   * This function makes the RocksDB#deleteFile Api synchronized by waiting
+   * for the deletes to happen.
+   * @param fileToBeDeleted File to be deleted.
+   * @throws RocksDBException In the underlying db throws an exception.
+   * @throws IOException In the case file is not deleted.
+   */
+  public void deleteFile(LiveFileMetaData fileToBeDeleted)
+      throws RocksDBException, IOException {
+    String sstFileName = fileToBeDeleted.fileName();
+    this.get().deleteFile(sstFileName);
+    File file = new File(fileToBeDeleted.path(), fileToBeDeleted.fileName());
+    ManagedRocksObjectUtils.waitForFileDelete(file, Duration.ofSeconds(60));
```

Comment:
Will the `IOException` thrown inside when exceeding 60s crash OM?

Reply:
Based on the current usage of this function, only SST filtering service will fail for the snapshot.

https://github.com/apache/ozone/blob/7a19afa71401e356cd86a99648e94278ea2850c8/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SstFilteringService.java#L193